invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/macros/detail.html

```diff
@@ -79,10 +87,11 @@ <h2>{{ add_description.type.title_l10n }} <span
 
 {% macro show_dates(dates) %}
   {% for date in dates %}
-    <dt class="ui tiny header">{{ date.type.title_l10n }}</dt>
+    {% set date_type_title = date.type.title_l10n if (date.type is defined and date.type.title_l10n is defined) else _('Unknown date type') %}
```

Comment:
How about using `get` since we should be getting serialized dicts (not really objects with attributes) as variables:

 ```suggestion
    {% set date_type_title = date.get('type', {}).get('title_l10n', _('Unknown date type')) %}
```